Lönn Support

catapillie opened the pull request:
Porting all Ahorn plugins to Lönn, making some improvements as well

catapillie committed:
Ported about 50% to Lönn

bigkahuna443 commented on Loenn/entities/battery.lua:
```diff
@@ -0,0 +1,10 @@
+    name = "battery"
+}
+
+battery.texture = "objects/FactoryHelper/batteryBox/battery00"
```
minor, but battery selection box is a bit large, might want to trim some in a selection function

catapillie replied:
now returning a 8x14 rectangle

bigkahuna443 commented on Loenn/entities/battery_box.lua:
```diff
@@ -0,0 +1,10 @@
+
+batteryBox.name = "FactoryHelper/BatteryBox"
+batteryBox.placements = {
+    name = "battery_box"
```
missing activation ID field + tooltip

bigkahuna443 commented on Loenn/entities/crate_spawner.lua:
```diff
@@ -0,0 +1,56 @@
+local normalColor = {1.0, 1.0, 1.0, 0.2}
+local randomColor = {1.0, 0.5, 1.0, 0.2}
+
+function crateSpawner.sprite(room, entity)
```
I kinda liked the arrow indicator on the Ahorn plugin to show it'd drop downwards, not a huge deal though

catapillie replied:
i would either need a sprite added to the mod exclusively for indicating that it drops down, or a way to draw arrows

bigkahuna443 replied:
I'll just make a note to add them after Lonn arrows happen lol

bigkahuna443 commented on Loenn/entities/debris.lua:
```diff
@@ -0,0 +1,40 @@
+    })
+end
+
+function debris.texture(room, entity)
```
Selection box is pretty large on these

catapillie replied:
returning a 18x18 rect now

bigkahuna443 commented on Loenn/entities/door_rusty.lua:
```diff
@@ -0,0 +1,13 @@
+    name = "door_rusty"
+}
+
+doorRusty.texture = "objects/FactoryHelper/doorRusty/metaldoor00"
```
Selection box is VERY large on these

catapillie replied:
fixed; matches lönn selection for doors

bigkahuna443 commented on Loenn/entities/electrified_wall.lua:
```diff
@@ -0,0 +1,78 @@
+    function handler.rectangle(room, entity, _)
+        return handler.selection(room, entity)
+    end
+    function handler.fillColor(room, entity)
```
These rendering functions make the spikes look almost identical to the vanilla lightning in Loenn which is confusing. Might want to render the spikes as well like in the Ahorn plugin

catapillie replied:
makes the 4 handlers way harder to condense into one file, but i'll work on that. could also just split them into four files like i did for pistons (except i might recondense piston plugins).

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,133 @@
+mods.FactoryHelper.name=FactoryHelper
```
This is the display name, so would change to "Factory Helper"

catapillie replied:
shows up as FactoryHelper in Ahorn. i have considered it though, should this be a general guideline for display names?

bigkahuna443 replied:
I think that would make sense, yeah

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,133 @@
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.delay=The amount if time before a new crate is spawned after the previous one in seconds.
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.isMetal=Whether the crates spawned are metallic or not. If `false` they will be wooden. Ignored if Is Random is true.
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.isRandom=Whether the spawned crates should have a random material. If set to `true` the `Is Metal` value will be ignored.
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.fromTop=If `true` they will spawn on top, otherwise they will spawn at the nearest side. Defautls to `true`.
```
"Defautls" -> "Defaults"

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,133 @@
+entities.FactoryHelper/ThrowBoxSpawner.placements.name.crate_spawner=Crate Spawner
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.activationId=String value of the entity's Activation Id. When a factory activator with this Id is turned on the entity will toggle state.
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.startActive=If `true` the entity will initially be active.
+entities.FactoryHelper/ThrowBoxSpawner.attributes.description.delay=The amount if time before a new crate is spawned after the previous one in seconds.
```
"The amount if time" -> "The amount of time"

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,133 @@
+entities.FactoryHelper/DashFuseBox.placements.name.left=Dash Fuse Box (Left)
+entities.FactoryHelper/DashFuseBox.placements.name.right=Dash Fuse Box (Right)
+entities.FactoryHelper/DashFuseBox.attributes.description.direction=The direction the fuse box is facing.
+entities.FactoryHelper/DashFuseBox.attributes.description.activationIds=A comma separated set of string Ids for all the Activation Ids this box should activate.
```
might want to edit the tooltip to mention that no activation ID will make the fuse box default to the broken state? or could maybe add a validator function to force one to be entered (unless someone uses these for aesthetics lol)

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,133 @@
+entities.FactoryHelper/DashNegator.attributes.description.startActive=If `true` the entity will initially be active.
+
+# Debris
+entities.FactoryHelper/KillerDebris.placements.name.bronze=Debris (Bronze)
```
why not "Killer Debris" like in the original plugin?

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,133 @@
+entities.FactoryHelper/PistonUp.attributes.description.pauseTime=The time the piston head waits in place after movement is done before moving again in seconds.
+entities.FactoryHelper/PistonUp.attributes.description.initialDelay=The time it takes before the piston first starts moving in seconds.
+entities.FactoryHelper/PistonUp.attributes.description.startActive=If `true` the entity will initially be active.
+entities.FactoryHelper/PistonUp.attributes.description.heated=Whether the pipe is heated or not.
```
Would be nice to make clear that "heated" = grabbing it will kill you

catapillie replied:
i'm just porting the tooltips, but since we "own" the mod we might as well improve stuff like this. + i'll back tooltips backwards too

bigkahuna443 reviewed (changes requested):
General comments:
1. Should maybe not add placements for every single possibility, really pads the list. Maybe condense pistons to non-heated placements, debris to non-attached, and placements with an "active/inactive" split to default active?
2. Would definitely condense fan + piston plugins as discussed in Discord. Repeated logic + tooltips makes it harder to maintain
3. If you were going alphabetically, you skipped BoomBox
4. Really nitpicky but I'd prefer `jump_thru_rusty.lua` -> `jumpthru_rusty.lua` :^)
5. Reminder to copy any tooltip changes made back to Ahorn

catapillie committed:
Minor tweaks & fixes

catapillie committed:
Renamed Jumpthrough Rusty Lönn plugin file name

bigkahuna443 commented on Loenn/entities/battery_box.lua:
```diff
@@ -0,0 +1,13 @@
+batteryBox.placements = {
+    name = "battery_box",
+    data = {
+        activationsIds = ""
```
`activationsIds` -> `activationIds`

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -44,7 +45,7 @@ entities.FactoryHelper/DashNegator.placements.name.inactive=Dash Negator (Inacti
 
-# Debris
+# Killer Debris
 entities.FactoryHelper/KillerDebris.placements.name.bronze=Debris (Bronze)
```
Will need to change the actual placement names too lol

catapillie replied:
My bad, that's my fault. Fixed.

catapillie committed:
Add more Lönn plugins, rename Debris to Killer Debris

catapillie committed:
Fixed wrong field name in battery box plugin

catapillie committed:
Port all triggers

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -132,3 +132,24 @@ entities.FactoryHelper/PistonRight.attributes.description.startActive=If `true`
+entities.FactoryHelper/PowerLine.description.attributes.activationId=String value of the entity's Activation Id. When a factory activator with this Id is turned on the entity will toggle state.
+entities.FactoryHelper/PowerLine.description.attributes.startActive=If `true` the entity will initially be active.
+entities.FactoryHelper/PowerLine.description.attributes.colorCode=Hex color code (example: ff0000 is red) for the power line's power color, in both active and inactive state. Inactive is a dimmer version of this hex color.
+entities.FactoryHelper/PowerLine.description.attributes.initialDelay=The time it takes for the power line to switch states after it is activated.
```
missing "in seconds" for `initialDelay` (also applies to Rusty Lamp)

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -132,3 +132,24 @@ entities.FactoryHelper/PistonRight.attributes.description.startActive=If `true`
 
+# Power Line
+entities.FactoryHelper/PowerLine.placements.name.power_line=Power Line
+entities.FactoryHelper/PowerLine.description.attributes.activationId=String value of the entity's Activation Id. When a factory activator with this Id is turned on the entity will toggle state.
```
For all of these you flipped "description" and "attributes" so none of them appear

bigkahuna443 commented on Loenn/entities/rust_berry.lua:
```diff
@@ -0,0 +1,26 @@
+
+function rustBerry.sprite(room, entity)
+    local button = drawableSprite.fromTexture("objects/FactoryHelper/rustBerry/berry_01", entity)
+    local case = drawableSprite.fromTexture("objects/FactoryHelper/rustBerry/gear_01", entity)
```
I think you copied over some variable names lol

bigkahuna443 commented on Loenn/entities/rusty_lamp.lua:
```diff
@@ -0,0 +1,53 @@
+            ["None"] = "None",
+            ["Flicker On"] = "FlickerOn",
+            ["Light Flicker"] = "LightFlicker",
+            ["Turn Off, Flicker On"] = "TurnOffFlickerOn"
```
There's a `FlickerOff` routine in the code that's not in the Ahorn plugin either. Should probably add them to both?

bigkahuna443 commented on Loenn/entities/power_line.lua:
```diff
@@ -0,0 +1,34 @@
+local powerLine = {}
```
These only render in-game if the nodes are placed at right angles, so movement should probably be limited somehow.

Also I asked about line visibility in the Loenn thread. I think these should either have the node lines always visible or just have some kind of connector drawn between them. Loenn has some line-drawing functions already (e.g. gondola nodes)

catapillie replied:
Added a preview rendering of the path that should be shown in game

bigkahuna443 commented on Loenn/triggers/spawn_steam_wall.lua:
```diff
@@ -0,0 +1,20 @@
+spawnSteamWall.name = "FactoryHelper/SpawnSteamWallTrigger"
+spawnSteamWall.fieldInformation = {
+    speed = {
+        minimumValue = 0.0
```
I tested this out with a negative value and it was totally functional, I feel like we might as well allow it lol (also applies to the steam wall speed trigger)

catapillie replied:
Thanks for testing, was unsure, so I added the minimum limit going off the field's name.

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -153,3 +153,35 @@ entities.FactoryHelper/RustyLamp.attributes.description.activationId=String valu
+
+# Steam Wall Speed Trigger
+triggers.FactoryHelper/SteamWallSpeedTrigger.placements.name.steam_wall_speed=Steam Wall Speed
+triggers.FactoryHelper/SteamWallSpeedTrigger.attributes.description.speed=Changes the speed multiplier a spawned wall movement. Has no effect if there is no steam wall currently spawned. Default speed is 22 pixels per second. A tile is 8 pixels wide.
```
`Changes the speed multiplier a spawned wall movement` ?

also I would set the attribute name to "Speed Multiplier" or something so it's clear you're not setting it directly (also applies to the spawn trigger)

you can do that w/o changing the field itself by setting `attributes.names.speed`

catapillie committed:
Add remaining entities

catapillie committed:
Merge horizontal and vertical fan plugins into one file

catapillie committed:
Merge all four directional piston plugins into one file

catapillie committed:
Render full piston sprites

catapillie committed:
Fixed typo in Ahorn trigger plugin file name

catapillie committed:
Apply small requested changes

catapillie committed:
Remove debug print call

catapillie committed:
Improve Powerline Lönn plugin rendering a lot

bigkahuna443 committed:
Cleanup

1. Culled placements. All placements now default to active.
2. Improved electrified wall rendering
3. Adjusted some placement names (e.g. spikes) so that they display/order better
4. Clean up lang file a bit, backport tooltip changes
5. Removed editable dropdown from wind tunnel to avoid confusion

catapillie committed:
Fix syntax error in Wind Tunnel Lönn plugin
